juju/model.py

```diff
@@ -31,6 +31,22 @@ def relation(self, relation_name):
             # ideally integrating the error catching with Juju's mechanisms.
             raise TooManyRelatedApps(relation_name, num_related, 1)
 
+    def relation_by_id(self, relation_name, relation_id):
```

Comment:
This would feel nicer if we align with `get_unit` and have it as `get_relation` I suspect.. even more considering it takes two parameters about the relation.